src/main/webapp/app/exam/manage/student-exams/student-exams.component.ts

```diff
@@ -322,6 +314,10 @@ export class StudentExamsComponent implements OnInit, OnDestroy {
     private setStudentExams(studentExams: any): void {
```

Comment:
I would suggest to reverse the if statement to avoid nesting
```ts
if (!studentExams) {
  return;
}
this.studentExams = studentExams;
// Other setter code
```